docs/source/rfc/on-chain-scripting.org

```diff
@@ -0,0 +1,571 @@
+would not be the case if we used =wasmtime=. 
+
+Reproducility in Rust is problematic as in the case of WASM.
+
+Forwards compatibility is guaranteed by the ELF executable format,
+provided the dynamic libraries retain their ABI.  In principle,
+because none of the system libraries should be linked against, this is
+perfect forwards compatibility.
+
+Relative popularity of native code versus WASM is difficult to judge.
+However, native code is ubiquitous, as almost all of the web
+infrastructure depends on the presence of well-optimised native code.
+I am willing to concede that despite several decades of head-start,
+the native code might not be orders of magnitude more popular than web
+assembly. I am, however unwilling to concede that native code is less
+popular owing to factors outlined below.
+
+Support for native code is the best of all binary standards. The
+entirety of server and consumer electronics infrastructure must be
+replaced in order for native code to be abandoned. WASM is not
+well-established at this point and its use is largely situational.
+
+Native support for SDKs varies but is usually as good or better than
+with WASM.  JavaScript can not be interpreted without a native
+runtime, with that said, in lieu of embedding a full v8 interpreter,
+one can use a project like [[https://github.com/vercel/pkg][pkg]] to produce self-contained quasi-native
+applications.  This could be done with WASM, but with WASM the
+infrastructure to make use of this binary would not be present. The
+JVM compatibility is fixed by adding a Java runtime environment, which
+can be done once, and used directly.  As an addded benefit there are
+no special tools needed to produce a compatible =.jar= file and this
+process is well-documented. The same goes for any language that
+targets native environments and with a natively executable run-time.
+The only limiting factor in this instance is the portability of the
+run-time and ease of exposing the Iroha native objects to the
+run-time, which itself can be made easy with the introduction of an
+API in the form of a RISC ISA. Exposing native libraries for dynamic
+linkage is a matter of practicality[fn:3], rather than necessity. 
+
+Ease of maintenance is less than that of WASM. No run-time is present,
+so unless the host system is anything but a POSIX-compliant system
+with ELF executables, and the smartcontracts link against non-portable
+functions in the standard c library, there is no maintenance to speak
+of.
+
+To summarise, a native platform library is by almost all criteria
+either a direct upgrade, or a minor improvement. The one instance in
+which there is significantly more work to be done is the question of
+confinement.  What can generously be called a minor inconvenience is
+the fact that running native libraries on heterogeneous networks,
+utilising multiple architectures has to come with a separate emulation
+layer.  Experience with running =x86= containers on Apple silicone
```

Comment:
Using fat binaries (or some other way to have the same code compiled for different architectures) may be a solution, though it makes the size and auditability even worse